clang.md

```diff
@@ -1,54 +1,76 @@
 # Compiling C/C++ to WebAssembly
 
-## Rolling your own compiler
+Many high level languages already support compilation to WebAssembly
+through the experimental LLVM backend. Unfortunately, it is a tedious
+and poorly documented process. This document aims to alleviate some
+of this tedium with a step-by-step tutorial to compile some basic C
+code to WAST format using LLVM, as well as provide instructions for building
+the toolchain.
 
-Clang has a WebAssembly target, though it is not easy to use currently. First, a custom build must be made.
+## Dependencies
+
+- LLVM + Clang: Must be built with the experimental WebAssebmly backend enabled
+- Binaryen: Needed to convert the `.s` output of LLVM's backend to WAST
+
+Instructions for installing these necessary components follow.
+
+## Install LLVM and Clang with the WebAssembly backend
+
+### From the repo
+
+First, clone the needed repositories:
 
-To build `clang`:
 ```sh
 git clone http://llvm.org/git/llvm.git
 cd llvm/tools
 git clone http://llvm.org/git/clang.git
 cd ../projects
 git clone http://llvm.org/git/compiler-rt.git
+```
+
+Then initialize CMake:
+
+```sh
 mkdir ../build
 cd ../build
 cmake -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly -DLLVM_TARGETS_TO_BUILD= ..
+```
+
+Lastly, call `cmake` as usual:
+
+```sh
 cmake --build .
-cd ../..
 ```
 
-This will take anything from 1 to 5 hours.
+At the end of the build process the binaries will be in `bin`. Feel free to add that to your `PATH` for ease of use.
+
+## Install Binaryen
+
+This one is much easier. Simply:
 
-To build `binaryen`:
 ```sh
 git clone https://github.com/WebAssembly/binaryen.git
 cd binaryen
 mkdir build
 cd build
 cmake ..
 cmake --build .
-cd ../..
 ```
 
-## Using this compiler
+CMake will also generate an `install` target if you want to actually install Binaryen on your system.
```

Comment:
How would you run this? `cmake install`? Could you please make that explicit here?